src/Http/Http.Extensions/gen/StaticRouteHandlerModel/InvocationOperationExtensions.cs

```diff
@@ -11,52 +12,100 @@ namespace Microsoft.AspNetCore.Http.RequestDelegateGenerator.StaticRouteHandlerM
 internal static class InvocationOperationExtensions
 {
     private const int RoutePatternArgumentOrdinal = 1;
-    private const int RouteHandlerArgumentOrdinal = 2;
+    public static readonly string[] KnownMethods =
+    {
+        "MapGet",
+        "MapPost",
+        "MapPut",
+        "MapDelete",
+        "MapPatch",
+        "Map",
+        "MapMethods",
+        "MapFallback"
+    };
 
     public static bool TryGetRouteHandlerMethod(this IInvocationOperation invocation, SemanticModel semanticModel, out IMethodSymbol? method)
     {
-        foreach (var argument in invocation.Arguments)
+        method = null;
+        if (invocation.TryGetRouteHandlerArgument(out var argument))
         {
-            if (argument.Parameter?.Ordinal == RouteHandlerArgumentOrdinal)
-            {
-                method = ResolveMethodFromOperation(argument, semanticModel);
-                return true;
-            }
+            method = ResolveMethodFromOperation(argument, semanticModel);
+            return true;
         }
-        method = null;
         return false;
     }
 
-    public static IArgumentOperation? GetRouteHandlerArgument(this IInvocationOperation invocation)
+    public static bool TryGetRouteHandlerArgument(this IInvocationOperation invocation, [NotNullWhen(true)]out IArgumentOperation? argumentOperation)
     {
-        foreach (var argument in invocation.Arguments)
+        argumentOperation = null;
+        if (invocation.Syntax.TryGetMapMethodName(out var methodName))
         {
-            if (argument.Parameter?.Ordinal == RouteHandlerArgumentOrdinal)
+            var routeHandlerArgumentOrdinal = methodName switch
+            {
+                "MapGet" or "MapPost" or "MapDelete" or "MapPut" or "MapPatch" => 2,
+                "Map" => 2,
+                "MapMethods" => 3,
+                "MapFallback" when invocation.Arguments.Length == 3 => 2,
```

Comment:
super nit:

```suggestion
                "MapFallback" when invocation.Arguments.Length is 3 => 2,
```

Reply:
Would it be better just to do `var routeHandlerArgumentOrdinal = invocation.Arguments.Length - 1`? It'd be weird for the handler not to be the last argument, and it would be one less thing to have to update if we add new Map methods.

Reply:
Yep, we can axe this switch.